sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClientIntegrationTest.java

```diff
@@ -638,6 +671,10 @@ void receiveAndComplete(MessagingEntityType entityType, boolean isSessionEnabled
         final String messageId = UUID.randomUUID().toString();
         final ServiceBusMessage message = getMessage(messageId, isSessionEnabled);
 
+        if (isSessionEnabled) {
+            this.receiver = receiverMono.block(TIMEOUT);
```

Comment:
We're not disposing of the receiver that was already set. This is some weird hack. If the flow is completely different, we should split test cases instead of shoehorning this in.

Additionally, you can do:

```java
var receiver = sessionReceiver.acceptSession().cache(e -> Duration.MAX_VALUE, e -> Duration.MIN_VALUE, e -> Duration.MIN_VALUE);
try {
StepVerifier.create(receiver.flatMap(e -> e.receive...))
    .

} finally {
    receiver.close();
}
```